python/cuml/cuml/manifold/umap.pyx

```diff
@@ -604,7 +604,7 @@ class UMAP(Base,
         self.target_n_neighbors = target_n_neighbors
         self.target_weight = target_weight
 
-        self.deterministic = random_state is not None
+        self._deterministic = random_state is not None
```

Comment:
Please revert this change as noted in https://github.com/rapidsai/cuml/pull/7004#discussion_r2210954329 .